src/model/TickDataSource.ts

```diff
@@ -1,47 +1,34 @@
 import { MongoDataSource } from 'apollo-datasource-mongodb'
-import { TickEditFilterType, TickType } from '../db/TickTypes'
+import type { DeleteResult } from 'mongodb'
+import mongoose from 'mongoose'
+
+import { TickEditFilterType, TickInput, TickType } from '../db/TickTypes'
 import { getTickModel } from '../db/index.js'
 
 export default class TickDataSource extends MongoDataSource<TickType> {
   tickModel = getTickModel()
 
   /**
-     * @param tick
-     * takes in a new tick
-     * @returns
-     * returns that new tick
-     */
-  async addTick (tick: TickType): Promise<any> {
-    if (tick === undefined || tick === null) {
```

Comment:
These checks are all irrelevant because `tick` is enforced to be non-null by typescript.